pkg/reconcilers/k3s/README.md

```diff
@@ -0,0 +1,18 @@
+# k3s 
+
+## Code design
+
+`pkg/k3s` package adopts a **component driven** design which implies the k3s code to be splitted into files name after k3s components:
+
+- `apiserver.go` k3s API server
+- `reconciler.go` k3s reconciler which expose `Reconcile` function that reconcile all k3s components
+- `service.go` k3s service
+
+Each Go file contains all methods related to its component.
+
+## Install k3s as a controlplane
+
+It is decided to use [k3s docker image](https://docs.k3s.io/advanced#running-k3s-in-docker) to ease the process avoiding to use Helm (as done on `vcluster`).
```

Comment:
I am not clear on what is the alternative that is being rejected here.

It _is_ clear that the chosen approach involves introducing and maintaining a lot of code. Do we really want to do this? To be fair, since abandoning Rudder Helm has lacked valuable functionality. It looks to me like we are taking on re-implementing Rudder (perhaps just some core part of it).